configs/openSUSE/pam-modules.toml

```diff
@@ -1,143 +1,458 @@
-    "pam_lastlog.so",
-    "pam_limits.so",
-    "pam_listfile.so",
-    "pam_localuser.so",
-    "pam_loginuid.so",
-    "pam_mail.so",
-    "pam_mkhomedir.so",
-    "pam_motd.so",
-    "pam_namespace.so",
-    "pam_nologin.so",
-    "pam_permit.so",
-    "pam_pwhistory.so",
-    "pam_rhosts.so",
-    "pam_rootok.so",
-    "pam_securetty.so",
-    "pam_selinux.so",
-    "pam_sepermit.so",
-    "pam_shells.so",
-    "pam_stress.so",
-    "pam_succeed_if.so",
-    "pam_tally2.so",
-    "pam_time.so",
-    "pam_timestamp.so",
-    "pam_tty_audit.so",
-    "pam_umask.so",
-    "pam_unix.so",
-    "pam_unix_acct.so",
-    "pam_unix_auth.so",
-    "pam_unix_passwd.so",
-    "pam_unix_session.so",
-    "pam_userdb.so",
-    "pam_warn.so",
-    "pam_wheel.so",
-    "pam_xauth.so",
-    # systemd
-    "pam_systemd.so",
-    # sssd
-    "pam_sss.so",
-    # pam_mktemp
-    "pam_mktemp.so",
-    # pam_csync
-    "pam_csync.so",
-    # pam_chroot
-    "pam_chroot.so",
-    # pam_snapper (bnc#815383)
-    "pam_snapper.so",
-    # pam_gdm (bsc#1004346)
-    "pam_gdm.so",
-    # pam_slurm (bsc#1007053)
-    "pam_slurm.so",
-    # pam_slurm_adopt (bsc#1116758)
-    "pam_slurm_adopt.so",
-    # pam_script (bsc#1039848)
-    "pam_script.so",
-    # pam_yubico (bsc#1087060)
-    "pam_yubico.so",
-    # pam_oath (bsc#1089114)
-    "pam_oath.so",
-    # pam_p11 (bsc#1123916)
-    "pam_p11.so",
-    # pam_cifscreds (bsc#1150527)
-    "pam_cifscreds.so",
-    # libpwquality (bsc#1150520)
-    "pam_pwquality.so",
-    # lxc (bsc#1150519)
-    "pam_cgfs.so",
-    # google-authenticator-libpam (bsc#1150524) - potential future removal candidate
-    "pam_google_authenticator.so",
-    # pam_u2f (bsc#1087061)
-    "pam_u2f.so",
-    # pam_kwallet (bsc#993806)
-    "pam_kwallet5.so",
-    # pam_dbus (bsc#1039709)
-    "pam_dbus.so",
-    # google-compute-engine (bsc#1146353)
-    "pam_oslogin_admin.so",
-    "pam_oslogin_login.so",
-    # fprintd (bsc#792095)
-    "pam_fprintd.so",
-    # mariadb (bsc#1163362)
-    "pam_user_map.so",
-    # oddjob (bsc#1169494)
-    "pam_oddjob_mkhomedir.so",
-    # cockpit (bsc#1169614)
-    "pam_cockpit_cert.so",
-    "pam_ssh_add.so",
-    # pam (bsc#1171564)
-    "pam_usertype.so",
-    # pam (bsc#1171563)
-    "pam_setquota.so",
-    # kanidm (bsc#1173387)
-    "pam_kanidm.so",
-    # GNOME malcontent parental control (bsc#1177974)
-    "pam_malcontent.so",
-    # pam (bsc#1171562)
-    "pam_faillock.so"
+[[FileDigestGroup]]
+package = "pam_krb5"
+type = "pam"
+note = "legacy: not audited"
+bug = ""
+nodigests = [
+    "/lib*/security/pam_krb5.so",
+    "/lib*/security/pam_krb5afs.so",
+]
+
+[[FileDigestGroup]]
+package = "ecryptfs-utils"
+type = "pam"
+note = "legacy: not audited"
+bug = ""
+nodigests = [
+    "/usr/lib*/security/pam_ecryptfs.so",
+]
+
+[[FileDigestGroup]]
+package = "gnome-keyring-pam"
+type = "pam"
+note = "legacy: not audited"
+bug = ""
+nodigests = [
+    "/lib*/security/pam_gnome_keyring.so",
```

Comment:
We currently seem to be in a migration from /lib/security to /usr/lib/security. To make this migration smooth it would make sense to whitelist both locations like:

`*/lib*/security/pam_gnome_keyring.so`

Since the check only triggers for paths are located either in /usr/lib/security or /lib/security we could even just whitelist "*/pam_gnome_keyring.so", am I right?

Due to the coupling of packages to whitelisting entries there is also no danger that other packages hijack a PAM module name.

Reply:
I renamed them now in 3f7d5cafb2254ea6e890c1ce92ad26eda9e09ccd